hal/armv7m/imxrt106x/hal.c

```diff
@@ -78,10 +82,13 @@ addr_t hal_vm2phym(addr_t addr)
 
 int hal_launch(void)
 {
+	time_t now;
+
 	syspage_save();
 
 	/* Give the LPUART transmitters some time */
-	timer_wait(100, TIMER_EXPIRE, NULL, 0);
+	now = hal_getTime();
+	while ((hal_getTime() - now) < 100);
```

Comment:
`now` is somewhat misleading, as it marks start, not current moment (marked by hal_getTime()).

I suggest changing name of this variable to e.g. `start`

Reply:
Ok. I have changed variable's name in hal and uart as well.